core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -506,6 +521,20 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
       }
     }
 
+    conf.get(EXECUTOR_DECOMMISSION_CLEANUP_INTERVAL).map { cleanupInterval =>
```

Comment:
IIUC, the decommissioned executor is supposed to shutdown et the end whether by the external services or itself (when decommissioned is considered to be done), right?

I understand this clean task is a guarantee for executor shutdown. But I think, for users, it's really hard to get a reasonable value of the conf here since they don't know what happens during decommissioning and what the approximate time it would take.

Besides, in the case of decommission coming from the external services, I think we'd better respect the `spark.executor.decommission.killInterval` here. This would prevent us from killing the executor prematurely when `spark.executor.decommission.cleanupInterval` is shorter than `spark.executor.decommission.killInterval`.

Reply:
I think users should be able to understand that block migrations are occurring, or if they aren't they probably won't be customizing this part of the configuration anyways. I can add a note in the config docs that this should be a relatively high value.